packages/dev/core/src/Materials/standardMaterial.ts

```diff
@@ -871,47 +871,17 @@ export class StandardMaterial extends PushMaterial {
         return "StandardMaterial";
     }
 
-    /**
-     * Specifies if the material will require alpha blending
-     * @returns a boolean specifying if alpha blending is needed
-     */
-    public override needAlphaBlending(): boolean {
-        if (this._disableAlphaBlending) {
-            return false;
-        }
-
-        return (
-            this.alpha < 1.0 ||
-            this._opacityTexture != null ||
-            this._shouldUseAlphaFromDiffuseTexture() ||
-            (this._opacityFresnelParameters && this._opacityFresnelParameters.isEnabled)
-        );
-    }
-
-    /**
-     * Specifies if this material should be rendered in alpha test mode
-     * @returns a boolean specifying if an alpha test is needed.
-     */
-    public override needAlphaTesting(): boolean {
-        if (this._forceAlphaTest) {
-            return true;
-        }
-
-        return this._hasAlphaChannel() && (this._transparencyMode == null || this._transparencyMode === Material.MATERIAL_ALPHATEST);
-    }
-
-    /**
-     * @returns whether or not the alpha value of the diffuse texture should be used for alpha blending.
-     */
     protected _shouldUseAlphaFromDiffuseTexture(): boolean {
         return this._diffuseTexture != null && this._diffuseTexture.hasAlpha && this._useAlphaFromDiffuseTexture && this._transparencyMode !== Material.MATERIAL_OPAQUE;
     }
 
-    /**
-     * @returns whether or not there is a usable alpha channel for transparency.
-     */
-    protected _hasAlphaChannel(): boolean {
-        return (this._diffuseTexture != null && this._diffuseTexture.hasAlpha) || this._opacityTexture != null;
+    protected override _hasAlpha(mesh: AbstractMesh): boolean {
+        return super._hasAlpha(mesh) || this._shouldUseAlphaFromDiffuseTexture() || this._opacityTexture != null;
+    }
+
+    /** @override */
+    public override needAlphaBlending(): boolean {
+        return super.needAlphaBlending() || this._opacityFresnelParameters?.isEnabled;

```

Comment:
As for the PBR material, the checks on `this._opacityTexture` and `this._shouldUseAlphaFromDiffuseTexture()` are missing, and `needAlphaTesting` would also need to be overriden to check `this._hasAlphaChannel()`.